src/Pages/CountryDetail.jsx

```diff
@@ -1,151 +1,172 @@
+                {" "}
+                {countryDetail.topLevelDomain}
+              </span>
+            </p>
+            <p>
+              Currencies:
+              <span className="opacity-60"> {countryDetail.currency}</span>
+            </p>
+            <p>
+              Languages:
+              <span className="opacity-60"> {countryDetail.languages}</span>
+            </p>
+          </div>
+
+          <div className="flex flex-wrap gap-3 lg:items-center lg:col-start-2 lg:row-span-2 lg:place-content-center">
+            <h2 className="text-lg mb-2">Border Countries:</h2>
+
+            {countryDetail.borders.map((border) => (
+              <Link
+                key={border.code}
+                className="bg-light-theme-elements dark:bg-dark-theme-elements shadow-md p-2"
+                to={`/Details/${border.name}`}
+              >
+                {border.name}
+              </Link>
+            ))}
+          </div>
+        </div>
+      </main>
+
+      <footer className="text-center text-balance p-4">
+        <p className="text-sm">
+          Challenge by{" "}
+          <a className="text-blue-700" href="frontendmentor.io">
+            Frontend Mentor.
+          </a>{" "}
+          coded by{" "}
+          <a className="text-blue-700" href="https://github.com/Mazz100">
+            Mazen Hassan.
+          </a>
+        </p>
+      </footer>
+    </div>
+  );
 }
 
-
 //Exporting an asyc function loader and use it inside react router path
 
 export const countryDetailsLoader = async ({ params }) => {
-    const { countryId } = params;
-
-    try {
-
-        const countryDetail = await fetch(`https://restcountries.com/v3.1/name/${countryId}?fullText=true&fields=flags,name,nativeName,population,region,subregion,capital,tld,currencies,languages,borders`)
-            .then(res => res.json())
-            .then(countriesData => countriesData.map(country => ({
-                flags: country.flags.svg,
-                flagAlt: country.flags.alt,
-                name: country.name.common,
-                nativeName: Object.entries(country.name.nativeName)
-                    //Mapping through nativeName values with unique key
-                    .map(([key, value]) => ` ${value.official}`)
-                    .join(', '),
-                population: country.population.toLocaleString(),
-                region: country.region,
-                subRegion: country.subregion,
-                topLevelDomain: country.tld,
-                capital: country.capital,
-                languages: Object.values(country.languages).join(', '),
-                capital: country.capital,
-                currency: Object.entries(country.currencies).map(([key, value]) =>
-                    `${value.name}`).join(', '),
-                borders: country.borders,
-            })))
-
-
-        const borderNamesQuery = countryDetail.map(country => country.borders).join(',') //alpha codes expect comma separator
```

Comment:
Here this doesn't work. In your version `countryDetail` is an array, so you map that and get the borders inside, what you'll end up with is a doubly nested array `[['FRA','DE']]` which when you call `join` on it, won't work at all.

I've solved it by having the raw country not be an array, but you could also have solved here by calling `.flat()` before `.join`